lib/concurrent-ruby/concurrent/executor/ruby_executor_service.rb

```diff
@@ -16,10 +16,19 @@ def initialize(*args, &block)
 
     def post(*args, &task)
       raise ArgumentError.new('no block given') unless block_given?
+      deferred_action = nil
       synchronize do
         # If the executor is shut down, reject this task
-        return handle_fallback(*args, &task) unless running?
-        ns_execute(*args, &task)
+        unless running?
+          deferred_action = fallback_action(*args, &task)
+          break
```

Comment:
`break` looks weird to me here, how about a simply if/else?
Also assuming `synchronize` returns the value of the block it could be `deferred_action = synchronize do`

Reply:
Fixed.